src/os/pl-cstack.c

```diff
@@ -437,17 +437,17 @@ print_trace(btrace *bt, int me)
 #endif
 	       ) &&
 	       addr2line(info.dli_fname, offset, buf, sizeof(buf)) )
-	    Sdprintf("  [%d] %s [%p]\n", i, buf, addr);
+	    Sdprintf("  [%zd] %s [%p]\n", i, buf, addr);
 	  else if ( info.dli_sname )
-	    Sdprintf("  [%d] %s(%s+%p) [%p]\n",
+	    Sdprintf("  [%zd] %s(%s+%lu) [%p]\n",
```

Comment:
This probably does not work.   I think addr2line expects hex notation.   So, it is probably better to cast the uintptr_t back to `void*` and use `%p`.    More in general, _long_ is almost always wrong unless the variable is declared `long`.  This rarely happens in SWI-Prolog because long in Win64 is 32 bit.   On this platform `uintptr_t` is the same as `long long`, so it should be `llu` there.   

Typically, we should be using `%zd` or `%p` for pointer-sized objects.   In the purist world this is still incorrect because `size_t` can be smaller than `uintptr_t` on systems with segmented memory.    As is, SWI-Prolog won't run on such systems anyway and I don't know about any actively supported hardware where this happens.

Reply:
Well, the value is a pointer difference, which we compute by casting both to `uintptr_t`.    Possibly would have been better to cast both to `char*`?    The problem is that aliases like `uintptr_t` are `long` on some platforms, but something else on others.     As it was, it was more portable as `uintptr_t` has the size of a pointer and `%p` expects a pointer.    The sizes match.   In theory there could be ABIs that pass pointers differently from integers though ...

In general it seems one need to be careful where the type comes from and what it could be on different platforms.   Then select the format specifier and make sure the passed argument is of matching type, possibly by casting it.    If you make sure the type used for the cast is wide enough, this should always work.

Reply:
The `t` specifier is for `ptrdiff_t` - casting to `char*` seems best as it gives the same result on all platforms.